src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/CreateSdkSymbolsLayout.cs

```diff
@@ -142,13 +143,23 @@ public Hashtable IndexAllSymbols()
 
                 var id = new BlobContentId(metadataReader.DebugMetadataHeader.Id);
                 string guid = $"{id.Guid:N}";
-                if (!string.IsNullOrEmpty(guid) && !allPdbGuids.ContainsKey(guid))
+                string debugId = GetDebugId(guid, file);
+                if (!string.IsNullOrEmpty(guid) && !allPdbGuids.ContainsKey(debugId))
                 {
-                    allPdbGuids.Add(guid, file);
+                    allPdbGuids.Add(debugId, file);
                 }
             }
 
             return allPdbGuids;
         }
+
+        /// <summary>
+        /// Calculates a debug Id from debug guid and filename. We use this as a key
+        /// in PDB hashtable. Guid is not enough due to collisions in several PDBs.
+        /// </summary>
+        private string GetDebugId(string guid, string file)
```

Comment:
// personal preference

Consider utilizing an expression bodied method here for brevity reasons.

Reply:
Will fix - good call.

Reply:
Fixed in https://github.com/dotnet/installer/pull/17533/commits/b4722e9197606b53cc3c29a1d13c80989e1c23b2